src/react-integration/hooks/useInvalidator.ts

```diff
@@ -0,0 +1,26 @@
+import { useContext, useCallback } from 'react';
+
+import { ReadShape, Schema } from '../../resource';
+import { DispatchContext } from '../context';
+
+/** Invalidate a certain item within the cache */
+export default function useInvalidator<
+Params extends Readonly<object>,
+Body extends Readonly<object> | void,
+S extends Schema
+>(selectShape: ReadShape<S, Params, Body>, params: Params | null) {
```

Comment:
can we move params to the function itself so this mirrors useFetcher?

Reply:
Hmm so would it be `invalidate({ id })`? No need for body I guess